### PR TITLE
Add leave faction console command

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Console Commands
 | fpvp_setfaction | {"Name" "ID"} | Yes | Set a players faction. |
 | fpvp_factioninvite | {"Name"} | No | Invite a player to your faction. (For leader and founder only) |
 | fpvp_closemenu | {} | No | Close the menu if it's stuck. |
+| fpvp_leavefaction | {} | No | Leave the faction you are currently in |
```
Comment (justinmuskopf) [inline README.md]:
```suggestion
| fpvp_leavefaction | {} | No | Leave the faction you are currently in. |
```

File: lua/cfcfactions/core/sv/sv_factionmanagment.lua [hunk 2]
```diff
@@ -8,6 +8,7 @@ local table = table
 local string = string
 local os = os
 local fpm = cfcFactions.fpm
+local Users = cfcFactions.Users
```
Comment (justinmuskopf) [inline lua/cfcfactions/core/sv/sv_factionmanagment.lua]:
```suggestion
local factioneers = cfcFactions.Users
```

File: lua/cfcfactions/core/sv/sv_factionmanagment.lua [hunk 3]
```diff
@@ -76,6 +77,13 @@ end
 
 concommand.Add( "fpvp_printpermissions", printFactionPermissions )
 
+-- Makes the player leave his faction
+local function leaveFaction( ply, cmd, args )
+    Users:RemoveUser( ply )
```
Comment (justinmuskopf) [inline lua/cfcfactions/core/sv/sv_factionmanagment.lua]:
```suggestion
    factioneers:RemoveUser( ply )
```

**Review by justinmuskopf — CHANGES_REQUESTED** [review]

File: lua/cfcfactions/core/sv/sv_factionmanagment.lua [hunk 4]
```diff
@@ -76,6 +77,13 @@ end
 
 concommand.Add( "fpvp_printpermissions", printFactionPermissions )
 
+-- Makes the player leave his faction
```
Comment (justinmuskopf) [inline lua/cfcfactions/core/sv/sv_factionmanagment.lua]:
```suggestion
-- Makes the player leave their faction
```

File: lua/cfcfactions/core/sv/sv_factionmanagment.lua [hunk 5]
```diff
@@ -76,6 +77,13 @@ end
 
 concommand.Add( "fpvp_printpermissions", printFactionPermissions )
 
+-- Makes the player leave their faction
+local function leaveFaction( ply, cmd, args )
+    factioneers:RemoveUser( ply )
+end
```
Comment (ghost) [inline lua/cfcfactions/core/sv/sv_factionmanagment.lua]: I think go ahead and check for permission "CanLeaveFaction" before actually leaving. Then it'll be good to merge

```
local function leaveFaction( ply, cmd, args )
    if fpm:hasPermission( ply, "CanLeaveFaction") then
        factioneers:RemoveUser( ply )
    end
end
```

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]
Looks good

**Review by justinmuskopf — APPROVED** [review]

**Pull request merged**